calculator/main.py

```diff
@@ -26,7 +28,18 @@ def perform_operation(calc, operation, params):
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
     """
-    pass
+    if operation not in calc['operations'].keys():
+        raise InvalidOperation("Given operation is invalid")
+    if not all([isinstance(param, float) or isinstance(param, int) for param in params]):
+        raise InvalidParams("Given params are invalid.")
+
+    result = reduce(calc['operations'][operation], [float(x) for x in params])
+
+    op_time = datetime.now()
+    op_time_str = '{0}-{1:02d}-{2:02d} {3:02d}:{4:02d}:{5:02d}'.format(op_time.year, op_time.month, op_time.day,
```

Comment:
Apparently, the default format for datetime.now() matches what we needed. Also, strftime is a great way to format datetime.